databases/week1/SQL_class_HomeWORK/sql_homeworkWeek1.sql

```diff
@@ -0,0 +1,70 @@
+USE hyf_lesson1;
+
+-- Find out how many tasks are in the task table
+SELECT COUNT(id)
+FROM task;
+
+-- Find out how many tasks in the task table do not have a valid due date
+SELECT COUNT(id)
+FROM task
+WHERE due_date IS NULL;
+
+-- Find all the tasks that are marked as done
+SELECT task.id, task.title, task.description, status.name
+FROM task
+INNER JOIN status ON status.id=task.status_id
+WHERE status.name="done";
+
+-- Find all the tasks that are not marked as done
+SELECT task.id, task.title, task.description, status.name
+FROM task
+INNER JOIN status ON status.id=task.status_id
+WHERE status.name!="done";
+
+-- Get all the tasks, sorted with the most recently created first
+SELECT *
+FROM task
+ORDER BY created DESC;
+
+-- Get the single most recently created task
+SELECT *
+FROM task
+ORDER BY created DESC
+Limit 0, 1;
+
+-- Get the title and due date of all tasks where the title or description contains database
+SELECT title, due_date
+FROM task
+WHERE description LIKE "%database%" OR title LIKE "%database%";
```

Comment:
In mysql the `LIKE` is not case insensitive. Unless you are positive you only want this word with this exact case, you could use `WHERE LOWER(description) LIKE "%database%" OR LOWER(title) LIKE "%database%";`

Reply:
I wasn't aware it is case sensitive. Thanks for letting me know